k8s/client.go

```diff
@@ -17,6 +18,7 @@ import (
 type Client struct {
 	client *groupVersionClient
 	schema resource.Schema
+	codec  resource.Codec
```

Comment:
We can't do that at the moment because the only decoder in the `groupVersionClient` is JSON, see my comments in the other thread as to why this isn't extended at the moment. So for right now, we explicitly only use a JSON codec.

I'd like to eventually move this to be capable of handling more, but there are some blockers to doing it quickly and easily.